src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py

```diff
@@ -80,91 +78,33 @@ def create_figure(self) -> Figure:
+                                   for latency in source.data[label]], 'latency')
+            fig.add_tools(
+                HoverKeysFactory.create_instance('stacked_bar', target_objects).create_hover())
+
+        # add legend (for each var in stacked bar)
+        legend_items = [(label, [bar]) for label, bar in zip(reversed_y_label, stacks)]
+        legend_items.reverse()
+        legend = Legend(items=legend_items, location='bottom_left',
+                        orientation='vertical', click_policy='mute')
         fig.add_layout(legend, 'below')
-        fig.legend.click_policy = 'mute'
 
         return fig
 
     @staticmethod
-    def _get_stacked_bar_data(
-        data: dict[str, list[int | float]],
-        y_labels: list[str],
-        xaxis_type: str,
-        x_label: str,
-    ) -> tuple[dict[str, list[float]],  list[float]]:
-        """
-        Calculate stacked bar data.
-
-        Parameters
-        ----------
-        data : dict[str, list[int]]
-            Source data.
-        y_labels : list[str]
-            Y axis labels that are Node/Topic name.
-        xaxis_type : str
-            Type of x-axis of the line graph to be plotted.
-            "system_time", "index", or "sim_time" can be specified.
-            The default is "system_time".
-        x_label : str
-            X-axis label of data dict.
-            "start time" or "index".
-
-        Returns
-        -------
-        dict[str, list[float]]
-            Stacked bar data.
-        list[float]
-            Width list of bars.
-
-        """
-        output_data: dict[str, list[float]] = {}
-        x_width_list: list[float] = []
-
-        # Convert the data unit to second
-        output_data = BokehStackedBar._updated_with_unit(data, y_labels, 1e-6)
-        output_data = BokehStackedBar._updated_with_unit(output_data, ['start time'], 1e-9)
-
-        # Calculate the stacked y values
-        output_data = BokehStackedBar._stacked_y_values(output_data, y_labels)
-
-        if xaxis_type == 'system_time':
-            # Update the timestamps from absolutely time to offset time
-            output_data[x_label] = BokehStackedBar._updated_timestamps_to_offset_time(
-                output_data[x_label])
-
-            x_width_list = BokehStackedBar._get_x_width_list(output_data[x_label])
-            half_width_list = [x / 2 for x in x_width_list]
-
-            # Slide x axis values so that the bottom left of bars are the start time.
-            output_data[x_label] = BokehStackedBar._add_shift_value(
-                output_data[x_label], half_width_list)
-        elif xaxis_type == 'sim_time':
-            raise NotImplementedError()
-        else:  # index
-            output_data[x_label] = list(range(0, len(output_data[y_labels[0]])))
-            x_width_list = BokehStackedBar._get_x_width_list(output_data[x_label])
-
-        return output_data, x_width_list
-
-    @staticmethod
-    def _updated_with_unit(
+    def updated_with_unit(
```

Comment:
Move this function to the StackedBarSource class.

Reply:
Fixed it in [7413d0f](https://github.com/tier4/CARET_analyze/pull/307/commits/7413d0f72a3ce8e0052c712da4b7db406081e5be).